src/coord/rmr/chan.h

```diff
@@ -21,6 +23,14 @@ void MRChannel_Push(MRChannel *chan, void *ptr);
  * Return NULL if the channel is empty and MRChannel_Unblock was called by another thread */
 void *MRChannel_Pop(MRChannel *chan);
 
+/* Pop an item with a timeout. Wait until there is an item, the channel is unblocked, or timeout expires.
+ * Parameters:
+ *   - chan: the channel to pop from
+ *   - abstime: absolute time (CLOCK_MONOTONIC) when the timeout expires. If NULL, behaves like MRChannel_Pop.
+ *   - timedOut: output parameter, set to true if the function returned due to timeout
+ * Returns: the popped item, or NULL if unblocked or timed out */
+void *MRChannel_PopWithTimeout(MRChannel *chan, const struct timespec *abstime, bool *timedOut);
```

Comment:
This should become the only `MRChannel_Pop` implementation.

Reply:
Should be done in a separate PR.